components/TemplateGrid/TemplateGridWrapper.vue

```diff
@@ -3,93 +3,120 @@
+                        :item="item"
+                        :toggle-item-method="toggleItem"
+                    />
+                    <template
+                        v-if="isConnectionsVisible"
+                        #connection>
+                        <div
+                            class="item-area__line"
+                            :style="connectionLineStyle(item)" />
+                    </template>
                 </TemplateGridItemArea>
             </TemplateGridItemsContainer>
         </TemplateGridContainer>
     </div>
 </template>
 
 <script>
-import { mapState, mapActions, mapGetters } from 'vuex';
-import {
-    getMaxChildRow,
-} from '~/model/tree/TreeCalculations';
+import { mapState, mapActions } from 'vuex';
+import { getMaxChildRow } from '~/model/tree/TreeCalculations';
 import TemplateGridPresentationLayer from '~/components/TemplateGrid/TemplateGridPresentationLayer';
 import TemplateGridItemsContainer from '~/components/TemplateGrid/TemplateGridItemsContainer';
 import TemplateGridContainer from '~/components/TemplateGrid/TemplateGridContainer';
 import TemplateGridGhostItem from '~/components/TemplateGrid/TemplateGridGhostItem';
 import TemplateGridItemArea from '~/components/TemplateGrid/TemplateGridItemArea';
-import TemplateGridHeader from '~/components/TemplateGrid/TemplateGridHeader';
-import CategoryTreeItem from '~/components/CategoryTree/CategoryTreeItem';
 
 export default {
-    name: 'CategoryTreeWrapper',
+    name: 'TemplateGridWrapper',
     components: {
         TemplateGridPresentationLayer,
         TemplateGridItemsContainer,
         TemplateGridContainer,
         TemplateGridGhostItem,
         TemplateGridItemArea,
-        TemplateGridHeader,
-        CategoryTreeItem,
+    },
+    props: {
+        isDraggingEnabled: {
+            type: Boolean,
+            default: false,
+        },
+        gridGap: {
+            type: Number,
+            default: 10,
+        },
+        isConnectionsVisible: {
+            type: Boolean,
+            default: true,
+        },
+        columns: {
+            type: Number,
+            required: true,
+        },
+        rowsHeight: {
+            type: Number,
+            required: true,
+        },
+        methodAfterDrop: {
+            type: Function,
+            default: () => {},
+        },
+        methodAfterRemove: {
+            type: Function,
+            default: () => {},
+        },
     },
     computed: {
-        ...mapState('tree', {
-            columns: state => state.treeLevels,
-            rowsHeight: state => state.rowsHeight,
-            rowsCount: state => state.rowsCount,
-            treeData: state => state.treeData,
+        ...mapState('gridDesigner', {
+            rows: state => state.rows,
+            gridData: state => state.gridData,
         }),
-        ...mapGetters('tree', [
-            'getChildrenLengthById',
-            'getExpandStateById',
-        ]),
+        filteredGridData() {
```

Comment:
filteredGridData does not tell you what exactly is in that data, as we can see there are some items smaller than something